extensions/redis-client/deployment/src/main/java/io/quarkus/redis/deployment/client/RedisClientProcessor.java

```diff
@@ -294,7 +294,8 @@ private static List<String> getRedisLoadScript(RedisClientBuildTimeConfig config
         if (config == null) {
             return List.of("import.redis");
         }
-        var scripts = config.loadScript;
+        // TODO @radcortez, this apparently returns a String now instead of a List<String>
+        var scripts = config.loadScript();
```

Comment:
You need to map it like this:
`Optional<List<@WithConverter(TrimmedStringConverter.class)String>> loadScript();`

`@WithConverter` now applies in the exact nested element type.

Reply:
That's what I did and the class cast error:

```java
    @ConfigDocDefault("import.redis in DEV, TEST ; no-file otherwise")
    Optional<List<@WithConverter(TrimmedStringConverter.class) String>> loadScript();
```

But it looks like the converter is somehow not applied:

```
Caused by: io.quarkus.runtime.configuration.ConfigurationException: Unable to find file referenced in 'quarkus.redis.redis-load-script=import/my-import.redis,  sample.redis'. Remove property or add file to your path.
	at io.quarkus.redis.deployment.client.RedisClientProcessor.preloadRedisData(RedisClientProcessor.java:263)
	at io.quarkus.redis.deployment.client.RedisClientProcessor.init(RedisClientProcessor.java:185)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

Reply:
It needs to be defined as:
`Optional<@WithConverter(TrimmedStringConverter.class) List<String>> loadScript();`

Sorry for misleading in the previous commented. Just pushed a fix.

Reply:
Mkay. TBH, that's odd, I would have expected the converter to be placed on its target rather than the container.

Reply:
https://github.com/smallrye/smallrye-config/pull/948